pkg/plugins/optional/helm/v1alpha/scaffolds/init.go

```diff
@@ -0,0 +1,453 @@
+	templatesmetrics "sigs.k8s.io/kubebuilder/v4/pkg/plugins/optional/helm/v1alpha/scaffolds/internal/templates/chart-templates/metrics"
+	templateswebhooks "sigs.k8s.io/kubebuilder/v4/pkg/plugins/optional/helm/v1alpha/scaffolds/internal/templates/chart-templates/webhook"
+)
+
+var _ plugins.Scaffolder = &initScaffolder{}
+
+type initScaffolder struct {
+	config config.Config
+
+	fs machinery.Filesystem
+
+	force bool
+}
+
+// NewInitHelmScaffolder returns a new Scaffolder for HelmPlugin
+func NewInitHelmScaffolder(config config.Config, force bool) plugins.Scaffolder {
+	return &initScaffolder{
+		config: config,
+		force:  force,
+	}
+}
+
+// InjectFS implements cmdutil.Scaffolder
+func (s *initScaffolder) InjectFS(fs machinery.Filesystem) {
+	s.fs = fs
+}
+
+// Scaffold scaffolds the Helm chart with the necessary files.
+func (s *initScaffolder) Scaffold() error {
+	log.Println("Generating Helm Chart to distribute project")
+
+	// Extract Images scaffolded with DeployImage to add ENVVAR to the values
+	imagesEnvVars := s.getDeployImagesEnvVars()
+
+	// Extract webhooks from generated YAML files (generated by controller-gen)
+	webhooks, err := extractWebhooksFromGeneratedFiles()
+	if err != nil {
+		return fmt.Errorf("failed to extract webhooks: %w", err)
+	}
+
+	scaffold := machinery.NewScaffold(s.fs,
+		machinery.WithConfig(s.config),
+	)
+
+	buildScaffold := []machinery.Builder{
+		&templates.HelmChart{},
+		&templates.HelmValues{
+			HasWebhooks:  len(webhooks) > 0,
+			Webhooks:     webhooks,
+			DeployImages: imagesEnvVars,
+			Force:        s.force,
+		},
+		&templates.HelmIgnore{},
+		&chart_templates.HelmHelpers{},
+		&manager.ManagerDeployment{
+			Force:        s.force,
+			DeployImages: len(imagesEnvVars) > 0,
+			HasWebhooks:  len(webhooks) > 0,
+		},
+		&templatescertmanager.Certificate{},
+		&templatesmetrics.MetricsService{},
+	}
+
+	if len(webhooks) > 0 {
+		buildScaffold = append(buildScaffold, &templateswebhooks.WebhookTemplate{})
+		buildScaffold = append(buildScaffold, &templateswebhooks.WebhookService{})
+	}
+
+	if err := scaffold.Execute(buildScaffold...); err != nil {
+		return fmt.Errorf("error scaffolding helm-chart manifests: %v", err)
+	}
+
+	// Copy relevant files from config/ to dist/chart/templates/
+	err = s.copyConfigFiles()
+	if err != nil {
+		return fmt.Errorf("failed to copy manifests from config to dist/chart/templates/: %v", err)
+	}
+
+	return nil
+}
+
+// getDeployImagesEnvVars will return the values to append the envvars for projects
+// which has the APIs scaffolded with DeployImage plugin
+func (s *initScaffolder) getDeployImagesEnvVars() map[string]string {
+	deployImages := make(map[string]string)
+
+	pluginConfig := struct {
+		Resources []struct {
+			Kind    string            `json:"kind"`
+			Options map[string]string `json:"options"`
+		} `json:"resources"`
+	}{}
+
+	const deployImageKey = "deploy-image.go.kubebuilder.io/v1-alpha"
```

Comment:
nit: Might be worth it to have [this](https://github.com/kubernetes-sigs/kubebuilder/blob/a9ee3909f7686902879bd666b92deec4718d92c9/pkg/cli/alpha/internal/generate.go#L45-L46) public in order to avoid hard-coding these values in different places.

Reply:
we would need make public from the plugin itself, not this file.
this file is only for the alpha command
If we could make plugin from the plugin than we can use in this file as well.

Reply:
It is a good point, but maybe is better to do in a follow up for all plugins

Reply:
I opened a issue for that: https://github.com/kubernetes-sigs/kubebuilder/issues/4309

Reply:
Kudos for opening an issue right away.